baremetal/metal3machine_manager.go

```diff
@@ -299,14 +299,6 @@ func (m *MachineManager) Associate(ctx context.Context) error {
 		return nil
 	}
 
-	config := m.Metal3Machine.Spec
-	err := config.IsValid()
-	if err != nil {
-		// Should have been picked earlier. Do not requeue
-		m.SetError(err.Error(), capierrors.InvalidConfigurationMachineError)
-		return nil
-	}
-
```

Comment:
I remember this being discussed, but this is not directly related to what this PR trying to achieve, is not it?

Reply:
It is not. It's essentially an alternative solution to #683. I can split it into a fix for the validate function and moving the validation to the webhooks if you prefer.

Reply:
hmm, no strong opinion on splitting & moving, but I wanted to make sure is, not to fix different issues with the same commit at least.

Reply:
also 684 will be redundant after this AFAIU

Reply:
you mean #683 @furkatgofurov7 ?